cpp/include/cuspatial/detail/algorithm/is_point_in_polygon.cuh

```diff
@@ -99,5 +100,96 @@ __device__ inline bool is_point_in_polygon(vec_2d<T> const& test_point, PolygonR
   return point_is_within;
 }
 
+/** @brief check if p3 is on the left side of the arc that is
+ * defined but p1 and p2
+ * @note: T can be float or double and T4 can be float4 or double4
+ * @param p1: first point of an arc
+ * @param p2: second point af an arc
+ * @param p3: a point to check
+ * @return bool
+ */
+template <typename T>
+__device__ bool is_left(vec_3d<T> const p1, vec_3d<T> const& p2, vec_3d<T> const& p3)
+{
+  vec_3d<T> po  = {0, 0, 0};
+  auto p1o      = po - p1;
+  auto p1p2     = p2 - p1;
+  auto p1p3     = p3 - p1;
+  auto p1oxp1p2 = cross_product(p1o, p1p2);
+  auto w        = dot(p1oxp1p2, p1p3);
+
+  return w > 0;
```

Comment:
Hmmm, I would have done `dot(cross(p1p2, p1p3), p3) > 0`. Seems more intuitive -- basically pointing right finger down vector p1p2, curling fingers toward p3, and then dot resulting vector with the vector from center of earth to p3 to see if it is positive magnitude.  But in any case, your code can be simplified (no need to subtract from zero).

```suggestion
  auto p1p2     = p2 - p1;
  auto p1p3     = p3 - p1;
  auto p1o_x_p1p2 = cross_product(-p1, p1p2);
  auto w        = dot(p1o_x_p1p2, p1p3);
  return w > 0;
```

Reply:
My preferred approach would be a clear one liner like this -- this matches my computer graphics experience much better.
```suggestion
  return dot(cross(p2-p1, p3-p1), p3) > 0;
```

Reply:
Updated as single liner.